src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModel.java

```diff
@@ -180,23 +182,47 @@ public Object getRoot() {
         return ROOT;
     }
 
-    private boolean hasContextChanged() {
-        Context newContext = ConfigHelper.getCurrentContext();
-        return newContext == null || context == null || !StringUtils.equals(context.getCluster(), newContext.getCluster()) || !StringUtils.equals(context.getUser(), newContext.getUser());
-    }
-
     @Override
-    public void onUpdate(ConfigWatcher source) {
+    public void onUpdate(ConfigWatcher source, Config config) {
         try {
-            if (ConfigHelper.isKubeConfigParsable() && hasContextChanged()) {
+            if (ConfigHelper.isKubeConfigParsable()) {
+              Context context = KubeConfigUtils.getCurrentContext(this.config);
+              Context newContext = KubeConfigUtils.getCurrentContext(config);
+              if (hasServerChanged(newContext, context)
```

Comment:
Maybe worth having these 2 method calls wrapped in a 
`hasContextChanged(config)`

for improved readability ?